### PR TITLE
More Phan Fixes

### PR DESCRIPTION
**Pull request by driusan**
This does another round of randomly fixes some of the smaller, low hanging fruit errors detected by phan.

### CODE REVIEW

File: php/libraries/Database.class.inc [hunk 1]
```diff
@@ -619,7 +619,7 @@ class Database
      */
     function selectRow($query, &$result)
     {
-        $this->select($query, $result);
+        $result = $this->pselect($query, []);
```
Comment (xlecours) [inline php/libraries/Database.class.inc]: `[]` => `array()`
Unless you want to start a new convention

File: php/libraries/Database.class.inc [hunk 2]
```diff
@@ -641,8 +641,7 @@ class Database
      */
     function selectOne($query)
     {
-        $result = null;
-        $this->select($query, $result);
+        $result = $this->pselect($query);
```
Comment (xlecours) [inline php/libraries/Database.class.inc]: this require a second argument

**Comment by xlecours** [comment]
The Travis is failing because of phpcs in htdocs/survey.php 
`42 | ERROR | Missing short description in doc comment`

File: php/libraries/Database.class.inc (hunk 1 above)
Comment (driusan) [reply php/libraries/Database.class.inc]: :(

File: php/libraries/Database.class.inc (hunk 2 above)
Comment (driusan) [reply php/libraries/Database.class.inc]: yes, I guess that's one of the error types which are still suppressed by phan.. I'll fix it.

File: htdocs/api/v0.0.1/APIBase.php [hunk 3]
```diff
@@ -199,7 +199,7 @@ function handleOPTIONS()
     {
         $this->Header(
             "Access-Control-Allow-Methods: ".
-            join($this->AllowedMethods, ",")
+            join(",", $this->AllowedMethods)
         );
```
Comment (PapillonMcGill) [inline htdocs/api/v0.0.1/APIBase.php]: Would it make more sense to use implode() in lieu of join() as PHP docs description is in implode() and join() page refer to implode() page.
Not a blocker, just a discussion

Comment (driusan) [reply htdocs/api/v0.0.1/APIBase.php]: Yes, but this is the minimum change to fix the "PhanParamSuspiciousOrder" error

**Comment by driusan** [comment]
This was included in #3544, but not auto-closed because the merge was squashed..

**Pull request closed without merging**